qfieldsync/core/cloud_api.py

```diff
@@ -182,19 +260,36 @@ def server_urls() -> List[str]:
             "https://localhost:8002/",
         ]
 
+    def username(self) -> Optional[str]:
+        if self.auth_method == CloudAuthMethod.CREDENTIALS:
+            return self.auth().config("username")
+        elif self.auth_method == CloudAuthMethod.SSO:
+            if self.current_username:
+                return self.current_username
+            else:
+                return None
```

Comment:
Isn't it the same if you just write `self.current_username`? I know it might be an empty string, but isn't it a worse error?